nullaway/src/test/java/com/uber/nullaway/NullAwayCoreTests.java

```diff
@@ -742,4 +742,26 @@ public void staticCallZeroArgsNullCheck() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void primitiveCasts() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "class Test {",
+            "    static void m() {",
+            "        Integer i = null;",
+            "        // BUG: Diagnostic contains: unboxing",
+            "        int i2 = (int) i;",
+            "        // this is fine",
+            "        int i3 = (int) Integer.valueOf(3);",
+            "        // try another type",
+            "        Double d = null;",
+            "        // BUG: Diagnostic contains: unboxing",
+            "        double d2 = (double) d;",
```

Comment:
I am not sure if we support `var` . But do we want to add a test case for ` var  v1 = (int) i` ?

Reply:
So in this PR, we add logic that looks at the `TypeCastTree` itself, independent of the surrounding context.  So it shouldn't matter whether it's in an assignment or not, and the `var` shouldn't matter either.  I'll add a couple of tests of casts in other contexts to make sure those cases work.

In terms of adding a test with `var`, since NullAway still supports Java 8 that is slightly painful.  Once we drop Java 8 support (hopefully soon) it'll be a lot easier to add such tests.